cms/djangoapps/contentstore/views/certificates.py

```diff
@@ -0,0 +1,380 @@
+                store.update_item(course, request.user.id)
+                course = _get_course_and_check_access(course_key, request.user)
+                print course.certificates['certificates']
+                return response
+        else:
+            return HttpResponse(status=406)
+
+
+@login_required
+@ensure_csrf_cookie
+@require_http_methods(("POST", "PUT", "DELETE"))
+def certificates_detail_handler(request, course_key_string, certificate_id):
+    """
+    JSON API endpoint for manipulating a course certificate via its internal identifier.
+    Utilized by the Backbone.js 'certificates' application model
+
+    POST or PUT
+        json: update the specified certificate based on provided information
+    DELETE
+        json: remove the specified certificate from the course
+    """
+    course_key = CourseKey.from_string(course_key_string)
+    store = modulestore()
+    with store.bulk_operations(course_key):
+        course = _get_course_and_check_access(course_key, request.user)
+        certificates_list = course.certificates.get('certificates', [])
+
+        match_index = None
+        match_cert = None
+        for index, cert in enumerate(certificates_list):
+            if certificate_id is not None:
+                if int(cert['id']) == int(certificate_id):
+                    match_index = index
+                    match_cert = cert
+
+        if request.method in ('POST', 'PUT'):
+            try:
+                new_certificate = CertificateManager.deserialize_certificate(course, request.body)
+            except CertificateValidationError as err:
+                return JsonResponse({"error": err.message}, status=400)
+
+            if match_cert:
+                certificates_list[match_index] = CertificateManager.to_json(new_certificate)
+            else:
+                certificates_list.append(CertificateManager.to_json(new_certificate))
+
+            store.update_item(course, request.user.id)
+            return JsonResponse(CertificateManager.to_json(new_certificate), status=201)
+
+        elif request.method == "DELETE":
+            if not match_cert:
+                return JsonResponse(status=404)
+
+            return CertificateManager.remove_certificate(
+                request=request,
+                store=store,
+                course=course,
+                certificate_id=certificate_id
+            )
+
+
+@login_required
+@ensure_csrf_cookie
+@require_http_methods(("POST", "PUT", "DELETE"))
+def signatory_detail_handler(request, course_key_string, certificate_id, signatory_id):
+    """
+    JSON API endpoint for manipulating a specific course certificate signatory via its internal identifier.
+    Utilized by the Backbone 'certificates' application.
+
+    DELETE
+        json: Remove the specified signatory from the specified certificate
+    """
+    course_key = CourseKey.from_string(course_key_string)
+    store = modulestore()
+    with store.bulk_operations(course_key):
```

Comment:
Not sure that you need `bulk_operations` here, seems to be only the course-load that uses it. Shouldn't matter, but when there are only fetches, I'm not certain that it is necessary...